OpenRA.Mods.Common/Traits/AttackMove.cs

```diff
@@ -137,16 +149,34 @@ public override void SelectionChanged(World world, IEnumerable<Actor> selected)
 
 		public override string GetCursor(World world, CPos cell, int2 worldPixel, MouseInput mi)
 		{
-			var prefix = mi.Modifiers.HasModifier(Modifiers.Ctrl) ? "assaultmove" : "attackmove";
+			var isAssaultMove = mi.Modifiers.HasModifier(Modifiers.Ctrl);
 
-			if (world.Map.Contains(cell) && subjects.Any())
+			var subject = subjects.FirstOrDefault();
+			if (subject.Actor != null)
 			{
-				var explored = subjects.First().Actor.Owner.Shroud.IsExplored(cell);
-				var blocked = !explored && subjects.Any(a => !a.Trait.Info.MoveIntoShroud);
-				return blocked ? prefix + "-blocked" : prefix;
+				var info = subject.Trait.Info;
+				if (world.Map.Contains(cell))
+				{
+					var explored = subject.Actor.Owner.Shroud.IsExplored(cell);
+					var cannotMove = subjects.FirstOrDefault(a => !a.Trait.Info.MoveIntoShroud).Trait;
+					var blocked = !explored && cannotMove != null;
+					if (isAssaultMove && !blocked)
+						return info.AssaultMoveCursor;
+					if (isAssaultMove && blocked)
+						return cannotMove.Info.AssaultMoveBlockedCursor;
+					if (!isAssaultMove && !blocked)
```

Comment:
By process of elimination, `!isAssaultMove` is by always true here, and `!isAssaultMove && blocked` below.

I suggest simplifying this to
```
if (isAssaultMove)
	return blocked ? cannotMove.Info.AssaultMoveBlockedCursor : info.AssaultMoveCursor;

return blocked ? cannotMove.Info.AttackMoveBlockedCursor : info.AttackMoveCursor;
```

Reply:
Thanks. That block got way out of hand.